app/src/main/kotlin/com/github/gotify/service/WebSocketService.kt

```diff
@@ -111,7 +113,41 @@ internal class WebSocketService : Service() {
         val intentFilter = IntentFilter()
         intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION)
 
-        picassoHandler.updateAppIds()
+        fetchAppIds(
+            onSuccess = { apps ->
+                picassoHandler.updateApps(apps)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    createNotificationChannels(apps)
+                }
+            },
+            onError = { picassoHandler.updateApps(listOf()) }
+        )
+    }
+
+    private fun fetchAppIds(
+        onSuccess: (apps: List<Application>) -> Unit,
+        onError: () -> Unit
+    ) {
+        ClientFactory.clientToken(settings.url, settings.sslSettings(), settings.token)
+            .createService(ApplicationApi::class.java)
+            .apps
+            .enqueue(
+                Callback.call(
+                    onSuccess = Callback.SuccessBody { apps ->
+                        onSuccess(apps)
+                    },
+                    onError = { onError() }
+                )
+            )
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun createNotificationChannels(apps: List<Application>) {
+        NotificationSupport.createChannels(
+            this,
+            (this.getSystemService(NOTIFICATION_SERVICE) as NotificationManager),
```

Comment:
I mean it's personal preference whether to produce it inside the method or hand it over.  
This version is copied from the previous PR.  
But since we changed that for another method signature, I'll change this here as well.